src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -432,18 +432,22 @@ public void Flush()
             this.configuration.TelemetryChannel.Flush();
         }
 
-        private string GetAssemblyVersion()
+        private string GetSdkVersion()
         {
 #if !CORE_PCL
-            return VersionPrefix + typeof(TelemetryClient).Assembly.GetCustomAttributes(false)
+            string versionStr = typeof(TelemetryClient).Assembly.GetCustomAttributes(false)
                     .OfType<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
+            
 #else
-            return VersionPrefix + typeof(TelemetryClient).GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
+            string versionStr = typeof(TelemetryClient).GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
 #endif
+
+            Version version = new Version(versionStr);
+            return VersionPrefix + version.ToString(3) + "-" + version.Revision;
```

Comment:
Why hyphen revision?
The version.ToString(3) should give you major.minor.revision so this will concatenate -revision on again.